main/src/ca/uwaterloo/flix/MutationTester.scala

```diff
@@ -0,0 +1,278 @@
+        flix.addSourceCode("testee",
+            s"""
+            |def main(): Unit \\ IO =
+            |    println(Test.one())
+            |def two(): Int32 = 2
+            |mod Fake {
+            |    pub def fake(): Int32 = 99
+            |}
+            |mod Test {
+            |    pub def one(): Int32 = 1
+            |    pub def ite(b: Bool): Int32 = if(b) 1 else 2
+            |}""".stripMargin)
+
+        flix.addSourceCode("tester",
+            s"""
+            |mod MainTest {
+            |    @test
+            |    def testConstant(): Bool =
+            |        Assert.eq(1, Test.one())
+            |    @test
+            |    def testIte(): Bool =
+            |       Assert.eq(1, Test.ite(true))
+            |}""".stripMargin)
+        val root = flix.check().unsafeGet
+        val start = System.nanoTime()
+        val root1 = mutate(root, testee)
+        val end = System.nanoTime() - start
+        println("time to generate mutations: " + end.toString)
+        runMutations(flix, root, root1)
+        /**
+        val result = root1.map(r => flix.codeGen(r).unsafeGet)
+        val tests = result.map(res => res.getTests)
+        */
+    }
+
+    def runMutations(flix: Flix, root: TypedAst.Root, mutatedDefs: List[(Symbol.DefnSym, List[TypedAst.Def])]): Unit = {
+        val totalStartTime = System.nanoTime()
+        var timeTemp = 0.0
+        var survivorCount = 0
+        var mutantCounter = 0
+        mutatedDefs.foreach(mut => {
+            val defs = root.defs
+            mutantCounter += mut._2.length
+            mut._2.foreach(mDef => {
+                val start = System.nanoTime()
+                val n = defs + (mut._1 -> mDef)
+                val newRoot = root.copy(defs = n)
+                val cRes = flix.codeGen(newRoot).unsafeGet
+                val testResults = cRes.getTests.values.forall(c =>
+                    try {
+                        c.run() match {
+                            case java.lang.Boolean.TRUE => true
+                            case _ => false
+                        }
+                    } catch {
+                        case _: Throwable => false
+                    }
+                )
+                timeTemp += (System.nanoTime() - start).toFloat / 1000000000
+                if (testResults) {
+                    survivorCount += 1
+                    println("mutation in " + mDef.sym.toString + " survived")
+                }
+            })
+        })
+        val totalEndTime = System.nanoTime() - totalStartTime
+        println("there where " + survivorCount.toString + " surviving mutations, out of " + mutantCounter.toString + " mutations")
+        println("average time to test a mutant: " + timeTemp/mutantCounter)
+        println("total time to test all mutants: " + totalEndTime.toFloat / 1000000000)
+    }
+    def mutate(root: TypedAst.Root, testee: String): List[(Symbol.DefnSym, List[TypedAst.Def])] = {
```

Comment:
I would suggest to follow the structure of the data:

- mutateRoot
- mutateDefs
- mutateDef

It will make it easier for you.